openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -2020,7 +2020,7 @@ def createForce(self, system, topology, nonbondedMethod=NoCutoff, nonbondedCutof
             for atom_mapping in atom_mappings:
                 for (atom_index, map_atom_index) in atom_mapping.items():
                     # Retrieve NB params for reference atom (charge not set yet)
-                    [charge, sigma, epsilon] = force.getParticleParameters(atom_index)
+                    charge, sigma, epsilon = force.getParticleParameters(map_atom_index)
```

Comment:
I agree that we could make the variable names more explicit. `atom_mapping` comes from `topology._reference_to_topology_atom_mappings` which is documented [here](_reference_to_topology_atom_mappings).

Maybe I can rename them `reference_molecule_atom_id` and `topology_atom_id` to make the code clearer?